src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,184 @@
+    private StorageDao storageDao;
+    private User user;
+
+    @BeforeEach
+    void setUp() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl(storageDao);
+        user = new User();
+        user.setLogin(DEFAULT_LOGIN);
+        user.setPassword(DEFAULT_PASSWORD);
+        user.setAge(DEFAULT_AGE);
+    }
+
+    @ AfterEach
+    void tearDown() {
+        storageDao.clear();
+    }
+
+    @Test
+    void checkLengthOfUsersLogin_Ok() {
+        user.setLogin(DEFAULT_LOGIN);
+        registrationService.register(user);
+        assertTrue(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void checkLengthOfUsersPassword_Ok() {
+        user.setPassword(DEFAULT_PASSWORD);
+        registrationService.register(user);
+        assertTrue(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void checkCorrectAge_Ok() {
+        user.setAge(DEFAULT_AGE);
+        registrationService.register(user);
+        assertTrue(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void addUserWhoAlreadyExist_NotOk() {
+        registrationService.register(user);
+
+        User duplicateUser = new User();
+        duplicateUser.setLogin(DEFAULT_LOGIN);
+        duplicateUser.setPassword(DEFAULT_PASSWORD);
+        duplicateUser.setAge(DEFAULT_AGE);
+
+        assertThrows(
+                InvalidUserDataException.class,
+                () -> registrationService.register(duplicateUser),
+                "Expected register() to throw exception, but it didn't"
+        );
+
+        int userCount = 0;
+        for (User user : Storage.people) {
+            if (user.getLogin().equals(DEFAULT_LOGIN)) {
+                userCount++;
+            }
+        }
+        assertEquals(1, userCount, "Duplicate user should not be registered");
+    }
+
+    @Test
+    void ifLoginLessThanSixCharacters_NotOk() {
+        user.setLogin(LESS_THAN_6_CHARACTERS);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertFalse(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void ifPasswordLessThanSixCharacters_NotOk() {
+        user.setPassword(LESS_THAN_6_CHARACTERS);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertFalse(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void ifAgeLessThan18_NotOk() {
+        user.setAge(UNDERAGE);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertFalse(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void testNullLogin_NotOk() {
+        user.setLogin(NULL_STRING);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertFalse(registrationService.isSuccesfullyRegistered(user));
```

Comment:
The test method name 'testNullLogin_NotOk' is good, but make sure to include a more informative exception message in the actual implementation to help understand the failure reason.